llvm/include/llvm/IR/IntrinsicsDirectX.td

```diff
@@ -38,7 +38,7 @@ def int_dx_typedBufferStore
                             [IntrWriteMem]>;
 
 def int_dx_updateCounter
-    : DefaultAttrsIntrinsic<[], [llvm_any_ty, llvm_i8_ty]>;
+    : DefaultAttrsIntrinsic<[llvm_i32_ty], [llvm_any_ty, llvm_i8_ty]>;
```

Comment:
I just realized that the intrinsic also needs an attribute `IntrInaccessibleMemOrArgMemOnly`. I am editing the same file in my PR so if you won't add it I will soon after.